src/modules/toDoList.js

```diff
@@ -0,0 +1,78 @@
+import ToDoListHelper from './toDoListHelper.js';
+
+class ToDoList {
+  constructor() {
+    this.tasks = JSON.parse(localStorage.getItem('datastore')) || [];
+  }
+
+  getTasks = () => this.tasks;
+
+  showList = () => {
+    const listContainer = document.getElementById('to-do-list');
+    listContainer.innerHTML = '';
+
+    this.getTasks().forEach((todo) => {
+      listContainer.innerHTML += `<li class='todo-item'>
+        <input type='checkbox' class="checkbox" id=${todo.index}/>
+        <input type='text' class='item-description' data-pos='${todo.index}' 
+        value='${todo.description}' disabled/>
+        <span class='modify fa fa-ellipsis-v fa_custom'></span>
+        <span class='trash-btn d-none fa fa-trash delete' data-index=${todo.index}></span>
+      </li>`;
+    });
+
+    if (this.getTasks().length > 0) {
+      ToDoListHelper.editElipsisEventHandler(this);
+      ToDoListHelper.deleteButtonEventHandler(this);
+      ToDoListHelper.editTextEventHandler();
+      ToDoListHelper.checkBoxEventHandler(this);
+    }
+  }
+
+  saveToStorage = () => {
+    localStorage.setItem('datastore', JSON.stringify(this.getTasks()));
+  }
+
+  saveTaskInList = (description, completed = false) => {
+    this.tasks.push({ description, completed, index: this.getTasks().length + 1 });
+    this.saveToStorage();
+  }
+
+  deleteTaskFromList = (clear = false, itemIndex = -1) => {
+    if (clear) {
+      this.tasks = this.getTasks().filter((task) => task.completed === false);
```

Comment:
- [ ] while everything works as it supposes to be, there's a minor issue (i think its a visual issue) when I mark a task as completed, it does change to true, but when I refresh, even so, it's true, the check box is not displayed as checked, kindly fix that 🙏 

_potential solution: add a check when generating the task if it's completed then you show the check box as checked_

![scroll](https://user-images.githubusercontent.com/89282221/215106646-74c7faf6-c3e8-4331-854d-71c0a914719d.gif)

Reply:
Hi @ichala, Thanks for the valuable feedback. I have made the changes in the checkbox functionality according to the suggestions. Now, it's working fine.